### PR TITLE
Kleisli Categories, Kleisli maps

### PR DESCRIPTION
**Pull request by TOTBWF**
# Description

This PR defines the Kleisli category of a monad, and proves that the category of Kleisli maps is weakly equivalent to the
Kleisli category.

There are also a couple of chores in here: I've fixed some naming inconsistencies with lemmas for fully faithful functors.
I've also refactored EM categories to be displayed.

## Checklist

Before submitting a merge request, please check the items below:

- [x] I've read [the contributing guidelines](https://github.com/plt-amy/1lab/blob/main/CONTRIBUTING.md).
- [x] The imports of new modules have been sorted with `support/sort-imports.hs` (or `nix run --experimental-features nix-command -f . sort-imports`).
- [x] All new code blocks have "agda" as their language.

If your change affects many files without adding substantial content, and
you don't want your name to appear on those pages (for example, treewide
refactorings or reformattings), start the commit message and PR title with `chore:`.

### CODE REVIEW

**Comment by Lavenza** [comment]
[Pull request preview](https://preview.1lab.dev/403)

File: src/Cat/Diagram/Monad/Kleisli.lagda.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    module EM = Cat.Reasoning (Eilenberg-Moore M)
+    module Free = Functor (Free-EM {M = M})
+
+  open M hiding (M)
+  open Cat.Reasoning C
+
+```
+-->
+
+# Kleisli map {defines="kleisli-morphism kleisli-map"}
```
Comment (ncfavier) [inline src/Cat/Diagram/Monad/Kleisli.lagda.md:51]:
```suggestion
# Kleisli maps {defines="kleisli-morphism kleisli-map"}
```

File: src/Cat/Diagram/Monad/Kleisli.lagda.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  Kleisli-maps .Precategory.idr _ =
+    lswizzle (sym (unit.is-natural _ _ _)) right-ident
+  Kleisli-maps .Precategory.idl _ =
+    cancell left-ident
+  Kleisli-maps .Precategory.assoc _ _ _ = monad! M
+```
+</details>
+
+Note that each Kleisli map $f : \cC(X,M(Y))$ can be extended
+to and $M$-algebra homomorphism between [[free $M$-algebras|free-algebra]]
```
Comment (ncfavier) [inline src/Cat/Diagram/Monad/Kleisli.lagda.md:90]:
```suggestion
to an $M$-algebra homomorphism between [[free $M$-algebras|free-algebra]]
```

File: src/Cat/Diagram/Monad/Kleisli.lagda.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  Kleisli-maps→Kleisli-is-eso : is-eso Kleisli-maps→Kleisli
+  Kleisli-maps→Kleisli-is-eso ((X , alg) , essentially-free) = do
+    A , free ← essentially-free
+    pure (A , (super-iso→sub-iso _ free))
+```
+
+Note that we **cannot** upgrade this weak equivalence to an equivalence when
+$\cC$ is univalent, as categories of Kleisli maps are not, in general,
+univalent. To see why, consider the monad on $\Sets$ that maps every
+set $\top$. Note that every hom set of the corresponding category of
```
Comment (ncfavier) [inline src/Cat/Diagram/Monad/Kleisli.lagda.md:165]:
```suggestion
set to $\top$. Note that every hom set of the corresponding category of
```

File: src/Cat/Diagram/Monad/Kleisli.lagda.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+    module M = Monad M
+    module MR = Cat.Functor.Reasoning M.M
+    module EM = Cat.Reasoning (Eilenberg-Moore M)
+    module Free = Functor (Free-EM {M = M})
+
+  open M hiding (M)
+  open Cat.Reasoning C
+```
+
+As shown in the previous section, category of Kleisli maps is weakly
```
Comment (ncfavier) [inline src/Cat/Diagram/Monad/Kleisli.lagda.md:210]:
```suggestion
As shown in the previous section, the category of Kleisli maps is weakly
```

**Review by ncfavier — APPROVED** [review]

**Comment by plt-amy** [comment]
> I've also refactored EM categories to be displayed.

:heart:

**Pull request merged**